### PR TITLE
FEATURE: add universal cancel all orders api helper

### PR DESCRIPTION
**Pull request by c9s**

### CODE REVIEW

File: pkg/strategy/dca2/strategy.go [hunk 1]
```diff
@@ -289,30 +291,29 @@ func (s *Strategy) CleanUp(ctx context.Context) error {
...
 	var werr error
 	for {
 		s.logger.Infof("checking %s open orders...", s.Symbol)
 
 		openOrders, err := retry.QueryOpenOrdersUntilSuccessful(ctx, session.Exchange, s.Symbol)
 		if err != nil {
-			s.logger.WithError(err).Errorf("CancelOrdersByGroupID api call error")
-			werr = multierr.Append(werr, err)
+			s.logger.WithError(err).Errorf("unable to query open orders")
+			continue
```
Comment (gx578007) [inline pkg/strategy/dca2/strategy.go:307]: Will this hit api rate limit?

File: pkg/util/tradingutil/cancel.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	for s := range symbolMap {
+		symbols = append(symbols, s)
+	}
+
+	return symbols
+}
+
+func CollectOpenOrders(ctx context.Context, ex types.Exchange, symbols ...string) ([]types.Order, error) {
```
Comment (gx578007) [inline pkg/util/tradingutil/cancel.go:104]: Where does this method get called?

File: pkg/util/tradingutil/cancel.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+				anyErr = err
+			}
+		}
+
+		if anyErr == nil {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("unable to cancel all orders: %+v", openOrders)
```
Comment (gx578007) [inline pkg/util/tradingutil/cancel.go:73]: How about adding `anyErr`?

File: pkg/strategy/dca2/strategy.go (hunk 1 above)
Comment (c9s) [reply pkg/strategy/dca2/strategy.go:307]: I think we have the protection in the exchange api layer

File: pkg/util/tradingutil/cancel.go (hunk 3 above)
Comment (c9s) [reply pkg/util/tradingutil/cancel.go:73]: Sure

File: pkg/util/tradingutil/cancel.go (hunk 2 above)
Comment (c9s) [reply pkg/util/tradingutil/cancel.go:107]: This will be used from xtri

**Review by kbearXD — APPROVED** [review]

**Pull request merged**